### PR TITLE
Shared FRC tag

### PR DESCRIPTION
**Pull request by wkurniawan07**
The beginning of shared tag.

### CODE REVIEW

**Comment by wkurniawan07** [comment]
Ready for debut, at last. Anyone willing to review?

**Comment by junhaoyap** [comment]
I'll look at it after merging some stuff

File: src/main/webapp/WEB-INF/tags/shared/feedbackResponseComment.tag [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+               <c:if test="${frc.editDeleteEnabledOnlyOnHover}">style="display: none;"</c:if>
+               <c:if test="${not frc.instructorAllowedToEdit}">disabled="disabled"</c:if>>
+                <span class="glyphicon glyphicon-pencil glyphicon-primary"></span>
+            </a>
+        </c:if>
+    </div>
+    <!-- do not add whitespace between the opening and closing tags -->
+    <div id="plainCommentText-${divId}" style="margin-left: 15px;">${frc.commentText}</div>
+    <c:if test="${frc.editDeleteEnabled}">
+        <form style="display: none;" id="responseCommentEditForm-${divId}" class="responseCommentEditForm">
```
Comment (junhaoyap) [inline src/main/webapp/WEB-INF/tags/shared/feedbackResponseComment.tag]: visibilityOptions.tag uses a div, is there a difference between this and that one? Looks kinda similar but I think there's some differences that I can't spot.

File: src/main/webapp/WEB-INF/tags/student/feedbackResults/response.tag [hunk 2]
```diff
@@ -17,13 +18,8 @@
...
             <ul class="list-group comment-list">                                                                           
-                <c:forEach items="${response.comments}" var="comment">                                                                            
-                    <li class="list-group-item list-group-item-warning" id="responseCommentRow-${comment.frcId}">
-                        <div id="commentBar-${comment.frcId}">
-                            <span class="text-muted">From: ${comment.giverEmail} [${comment.createdAt}] ${comment.editedAtText}</span>
-                        </div>
-                        <div id="plainCommentText-${comment.frcId}" style="margin-left: 15px;">${comment.commentText}</div>
-                    </li>
+                <c:forEach items="${response.comments}" var="comment">
+                    <shared:feedbackResponseComment frc="${comment}" />
```
Comment (junhaoyap) [inline src/main/webapp/WEB-INF/tags/student/feedbackResults/response.tag]: This seems to add a lot of elements to the unodered list, was response.tag previously unused?

**Comment by junhaoyap** [comment]
Other changes are ok, I'm just puzzled over why one uses form and other uses div yet they both look so similar. Also, the part where a lot of elements are added, unless it was previously unused there should be HTML changes right?

File: src/main/webapp/WEB-INF/tags/student/feedbackResults/response.tag (hunk 2 above)
Comment (wkurniawan07) [inline src/main/webapp/WEB-INF/tags/student/feedbackResults/response.tag]: It is. What you saw up there is the simplest version of FRC. No icons or anything whatsoever.

**Comment by wkurniawan07** [comment]
Hmmm where exactly do I add lot of elements?

**Comment by junhaoyap** [comment]
Previously it was just a few list items then now it's the whole tag, doesn't that add a lot of elements?

**Comment by junhaoyap** [comment]
Oh I see, is it because all the if clauses are false in the simplest form?

File: src/main/webapp/WEB-INF/tags/shared/feedbackResponseComment.tag (hunk 1 above)
Comment (wkurniawan07) [inline src/main/webapp/WEB-INF/tags/shared/feedbackResponseComment.tag]: Another testament on how unstandardized things are, right?
`visibilityOptions.tag` belongs to Comments, not FRCs. I haven't reached the point where I want to standardize both FRC and Comment yet.

**Comment by wkurniawan07** [comment]
Nah, a huge portion of those whole tag aren't rendered. You know my technique, just render certain parts of the tag depending on the page it's used.

File: src/main/webapp/WEB-INF/tags/shared/feedbackResponseComment.tag (hunk 1 above)
Comment (junhaoyap) [inline src/main/webapp/WEB-INF/tags/shared/feedbackResponseComment.tag]: Ahha I see, ok then, perhaps we should standardize it some day, but not today, we have a timeline to meet :p

**Comment by junhaoyap** [comment]
Ok got it, let's get it into JSTL-Dev so we are one more step forward into completing it!

**Pull request merged**